File: python/2015/002/part_one.py
import common

if __name__ == "__main__":
    for input_file in common.inputs:
        result = 0

        for line in common.read_file(input_file):
            l, w, h = line.split("x")
            l, w, h = int(l), int(w), int(h)

            area = 2 * l * w + 2 * w * h + 2 * h * l
            sides = [w * l, w * h, l * h]

            result += area + min(sides)

        print(result)

File: kotlin/src/main/kotlin/pt/dmb/adventofcode/y2015/p02/PartOne.kt
package pt.dmb.adventofcode.y2015.p02

import pt.dmb.adventofcode.common.Utilities

class PartOne(
    private val length: Int,
    private val width: Int,
    private val height: Int,
) {
    fun area() = 2 * length * width + 2 * width * height + 2 * height * length
    fun minSide() = listOf(length * width, width * height, height * length).min()
}

fun main() {
    var result = 0

    Utilities.readInputFile(2015, 2, scope = "main").forEach { line ->
        val splits = line.split("x")

        val length = splits[0].toInt()
        val width = splits[1].toInt()
        val height = splits[2].toInt()

        val solver = PartOne(length, width, height)

        result += solver.area() + solver.minSide()
    }

    println(result)
}
